src/main/java/edu/harvard/iq/dataverse/globus/TaskMonitoringServiceBean.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
+ * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
+ */
+package edu.harvard.iq.dataverse.globus;
+
+import edu.harvard.iq.dataverse.settings.SettingsServiceBean;
+import edu.harvard.iq.dataverse.util.SystemConfig;
+import jakarta.annotation.PostConstruct;
+import jakarta.annotation.Resource;
+import jakarta.ejb.EJB;
+import jakarta.ejb.Singleton;
+import jakarta.ejb.Startup;
+import jakarta.enterprise.concurrent.ManagedScheduledExecutorService;
+import java.io.File;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.FileHandler;
+import java.util.logging.Logger;
+
+/**
+ * 
+ * This Singleton monitors ongoing Globus tasks by checking with the centralized
+ * Globus API on the status of all the registered ongoing tasks. 
+ * When a successful completion of a task is detected, the service triggers
+ * the execution of the associated tasks (for example, finalizing adding datafiles
+ * to the dataset on completion of a remote Globus upload). When a task fails or 
+ * terminates abnormally, a message is logged and the task record is deleted 
+ * from the database. 
+ * 
+ * @author landreev
+ */
+@Singleton
+@Startup
+public class TaskMonitoringServiceBean {
+    private static final Logger logger = Logger.getLogger("edu.harvard.iq.dataverse.globus.TaskMonitoringServiceBean");
+    
+    @Resource
+    ManagedScheduledExecutorService scheduler;
+    
+    @EJB
+    SystemConfig systemConfig;
+    @EJB
+    SettingsServiceBean settingsSvc;
+    @EJB 
+    GlobusServiceBean globusService;
+    
+    private static final SimpleDateFormat logFormatter = new SimpleDateFormat("yyyy-MM-dd'T'HH-mm-ss");
+    
+    @PostConstruct
+    public void init() {
+        if (systemConfig.isGlobusTaskMonitoringServer()) {
+            logger.info("Starting Globus task monitoring service");
+            int pollingInterval = SystemConfig.getIntLimitFromStringOrDefault(
+                settingsSvc.getValueForKey(SettingsServiceBean.Key.GlobusPollingInterval), 60);
```

Comment:
Longer default maybe, even 10 times longer (or more). I can see cases where someone may want to set it to a minute (e.g. testing) but for normal hours-long transfers, seems like wasted effort.